tests/test_bookmarks_metafields.py

```diff
@@ -0,0 +1,221 @@
+"""
+Test tap sets a bookmark and respects it for the next sync of a stream
+"""
+from datetime import datetime as dt, timezone
+
+from dateutil.parser import parse
+
+from tap_tester import menagerie, runner, LOGGER
+from base import BaseTapTest
+
+class BookmarkMetafieldsTest(BaseTapTest):
+    """Test tap sets a bookmark and respects it for the next sync of a stream"""
+    @staticmethod
+    def name():
+        return "tap_tester_shopify_bookmark_metafields_test"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.start_date = '2025-01-01T00:00:00Z'
+
+    def min_max_bookmarks_by_stream(self, sync_records):
+        """
+        Return the maximum value for the replication key for each stream
+        which is the bookmark expected value.
+
+        Comparisons are based on the class of the bookmark value. Dates will be
+        string compared which works for ISO date-time strings
+        """
+        max_bookmarks = {}
+        min_bookmarks = {}
+        metafields_shop = []
+        metafields_order = []
+        metafields_product = []
+        metafields_customer = []
+        metafields_collection = []
+        for stream, batch in sync_records.items():
```

Comment:
It seems that in the future, we may get more types of metafields in response, so I think it would be better to use a single dict instead of separate lists here.